### PR TITLE
Calculator events

### PR DESCRIPTION
**Pull request by dev-mamun**
**In This PR:**
<hr>

- Created a new directory called logic.
- Copied the files calculate.js and operate.js into logic folder.
- Installed [big.js](https://www.npmjs.com/package/big.js) using NPM.
- Imported the files in Calculator.js.
- Implemented click event handlers using the math logic from calculate.js and operate.js.

### CODE REVIEW

File: src/components/Buttons.js [hunk 1]
```diff
@@ -4,20 +4,200 @@
...
+              className="cal-btn"
+              onClick={() => setCalObj({ ...calObj, next: parseInt(`${calObj.next}`, 10) || 9 })}
+            >
+              9
+            </button>
+          </div>
+          <div className="btn-col">
+            <button type="button" className="cal-btn" onClick={() => setCalObj(calculate({ ...calObj }, '*'))}>
+              *
+            </button>
```
Comment (nahnahdev) [inline src/components/Buttons.js]: - [x] Great job so far. However, I noticed that when I clicked on this button, It threw the error message below. So kindly fix this :100: 
- Error message
image

File: src/components/Buttons.js [hunk 2]
```diff
@@ -4,20 +4,200 @@
...
+          </div>
+          <div className="btn-col">
+            <button
+              type="button"
+              className="cal-btn"
+              onClick={() => setCalObj(calculate({ ...calObj }, '.'))}
+            >
+              .
+            </button>
```
Comment (nahnahdev) [inline src/components/Buttons.js]: - Whenever I tried to click on this button, It also threw an error. So kindly fix this as well :+1: 
- Error Message 
image

File: src/App.js [hunk 3]
```diff
@@ -1,14 +1,11 @@
...
 function App() {
   return (
-    <Container>
-      <div className="App">
-        <Calculator />
-      </div>
-    </Container>
+    <div className="App">
+      <Calculator />
+    </div>
```
Comment (nahnahdev) [inline src/App.js:8]: - [x] Lastly, I noticed that after I clicked on a number and I wanted to click on another number, they all didn't work except `7`. But after I use one of the symbols that is when I will be able to click on another number and even after clicking on the number, I can't click on any other number except `7`. But as you know, any calculator should be able to process a variety of computations with various numbers so kindly fix this as well. I have attached a video reference below so you have a look :+1: 
- [x] You can also see from the video below that your calculator doesn't have the number `2` so kindly this as  well :+1: 

- Video Reference
[Screencast from 03-29-2023 05:19:58 PM.webm](https://user-images.githubusercontent.com/102699958/228618242-6a516494-5bd6-40c7-9d29-21f2d53d14b4.webm)

**Review by nahnahdev — CHANGES_REQUESTED** [review]
## Project Status: CHANGES REQUIRED :recycle: 
Hi @dev-mamun , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- No linter errors :heavy_check_mark: 
- Good use of GitFlow :heavy_check_mark: 
- Professional README :heavy_check_mark: 

## Required Changes ♻️
- [x] I noticed that in your README, you have a `Live Demo` link that leads to an error page. Since you haven't deployed your project yet, I will suggest that you just put `N/A` instead of including a false link. :+1: eg.

```
## 🚀 Live Demo <a name="live-demo"></a>

- N/A
``` 
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Reem-lab — COMMENTED** [review]
Hi @dev-mamun, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that you have a live demo section in your README file and to have a professional README file you should remove this section till you have your app deployed so you can just add the live demo section in case you have deployed link for your project is present. .✅❗💯🤩✨✨

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi @dev-mamun,

#### Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

### Congratulations! 🎉 

giphy (1)

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**